### ISSUE TITLE
What does the model.meta store?

### ISSUE THREAD

**Issue by BingoTiger** [davidsandberg/facenet#110]
Hello, these days we are trying to change your code to make it run on 4 gpus. We have finished the part of training, but there are some problems while vilidating. After using 4 gpus to do the trianing, model.meta stores the information about gpus,for example
these are some information in model.meta, you can see it stores which tower uses which gpu

wtower_0/gradients/tower_0/InceptionResnetV1/Repeat_2/block8_2/Branch_0/Conv2d_1x1/convolution_grad/Conv2DBackpropFilterConv2DBackpropFilter0tower_0/InceptionResnetV1/Repeat_2/block8_1/Relujtower_0/gradients/tower_0/InceptionResnetV1/Repeat_2/block8_2/Branch_0/Conv2d_1x1/convolution_grad/Shape_1tower_0/gradients/AddN_7/device:GPU:0*(

So when we use the validate_on_lfw.py, there is an erroe that can't find gpu.
I want to know what does the model.meta store. I know that we use saver.save(sess,checkpoint_path,global_step=step) to store sess, but I don't know which variable in session stores the information about device. Is there anyway to ignore the device information when store or validate?
We can send our changed code about training to you if neccessary.
I hope you can understand my problems.
Sorry for my poor English.
Thanks so much.

**Comment by BingoTiger** [comment]
What's more, in validate_on_lfw.py, you don't reload the inception_resnet_v1.py, so does model.meta store the information about the network?
Thanks.

**Comment by davidsandberg** [comment]
Hi @BingoTiger ,
The metagraph file contains the computational graph (without any parameter values). Some more details can be found [here](http://stackoverflow.com/questions/36195454/what-is-the-tensorflow-checkpoint-meta-file). I think the problem you have is addressed [here](https://www.tensorflow.org/versions/master/how_tos/meta_graph/).

Import a graph with preset devices.
Sometimes an exported meta graph is from a training environment that the importer doesn't have. For example, the model might have been trained on GPUs, or in a distributed environment with replicas. When importing such models, it's useful to be able to clear the device settings in the graph so that we can run it on locally available devices. This can be achieved by calling import_meta_graph with the clear_devices option set to True.

```
with tf.Session() as sess:
  new_saver = tf.train.import_meta_graph('my-save-dir/my-model-10000.meta',
      clear_devices=True)
  new_saver.restore(sess, 'my-save-dir/my-model-10000')
  ...
```

**Comment by BingoTiger** [comment]
ok, I will read these links, thanks
